homeassistant/components/sensor/nsw_fuel_station.py

```diff
@@ -0,0 +1,175 @@
+from homeassistant.components.light import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['nsw-fuel-api-client==1.0.10']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_STATION_ID = 'station_id'
+ATTR_STATION_NAME = 'station_name'
+
+CONF_STATION_ID = 'station_id'
+CONF_FUEL_TYPES = 'fuel_types'
+CONF_ALLOWED_FUEL_TYPES = ["E10", "U91", "E85", "P95", "P98", "DL",
+                           "PDL", "B20", "LPG", "CNG", "EV"]
+CONF_DEFAULT_FUEL_TYPES = ["E10", "U91"]
+CONF_ATTRIBUTION = "Data provided by NSW Government FuelCheck"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.positive_int,
+    vol.Optional(CONF_FUEL_TYPES, default=CONF_DEFAULT_FUEL_TYPES):
+        vol.All(cv.ensure_list, [vol.In(CONF_ALLOWED_FUEL_TYPES)]),
+})
+
+MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(hours=1)
+
+NOTIFICATION_ID = 'nsw_fuel_station_notification'
+NOTIFICATION_TITLE = 'NSW Fuel Station Sensor Setup'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the NSW Fuel Station sensor."""
+    from nsw_fuel import FuelCheckClient
+
+    station_id = config[CONF_STATION_ID]
+    fuel_types = config[CONF_FUEL_TYPES]
+
+    client = FuelCheckClient()
+    station_data = StationPriceData(client, station_id)
+    station_data.update()
+
+    if station_data.error is not None:
+        message = (
+            'Error: {}. Check the logs for additional information.'
+        ).format(station_data.error)
+
+        hass.components.persistent_notification.create(
+            message,
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+        return
+
+    available_fuel_types = station_data.get_available_fuel_types()
+
+    add_devices([
+        StationPriceSensor(station_data, fuel_type)
+        for fuel_type in fuel_types
+        if fuel_type in available_fuel_types
+    ])
+
+
+class StationPriceData(object):
+    """An object to store and fetch the latest data for a given station."""
+
+    def __init__(self, client, station_id: int) -> None:
+        """Initialize the sensor."""
+        self.station_id = station_id
+        self._client = client
+        self._data = None
+        self._reference_data = None
+        self.error = None
+        self._station_name = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Update the internal data using the API client."""
+        from nsw_fuel import FuelCheckError
+
+        if self._reference_data is None:
+            try:
+                self._reference_data = self._client.get_reference_data()
+            except FuelCheckError as exc:
+                self.error = str(exc)
+                _LOGGER.error(
+                    'Failed to fetch NSW Fuel station reference data',
+                    exc_info=exc)
```

Comment:
Just pass the exception as a string formatting parameter to the logging string message. Don't use `exc_info` kwarg.
```py
_LOGGER.error("bla bla %s", exc)
```